perllib/FixMyStreet/DB/Result/ModerationOriginalData.pm

```diff
@@ -139,26 +139,35 @@ sub compare_coords {
 sub compare_photo {
     my ($self, $other) = @_;
 
-    my $old = $self->photo || '';
-    my $new = $other->photo || '';
-    return '' if $old eq $new;
-
-    $old = [ split /,/, $old ];
-    $new = [ split /,/, $new ];
+    my $old = $self->photos;
+    my $new = $other->photos;
 
-    my $diff = Algorithm::Diff->new( $old, $new );
+    my (%old_photo_ids_set, %new_photo_ids_set);
     my (@added, @deleted);
-    while ( $diff->Next ) {
-        next if $diff->Same;
-        push @deleted, $diff->Items(1);
-        push @added, $diff->Items(2);
+
+    foreach (@$old) {
```

Comment:
All this code is fine, just some perl-isms you might want to know about to use here or in future:

A loop that is setting a hash based upon an array, like here, can be written along with its definition as:
    `my %old_photo_ids_set = map { $_->{id} => 1 } @$old;`
You'll see that used a few places in the code.